runtime/interpreter/interpreter.go

```diff
@@ -1881,6 +1897,33 @@ func (interpreter *Interpreter) compositeDestructorFunction(
 	}
 }
 
+func (interpreter *Interpreter) defaultFunctions(
+	members *ast.Members,
+	lexicalScope *VariableActivation,
+) map[string]FunctionValue {
+
+	functionDeclarations := members.Functions()
+	functionCount := len(functionDeclarations)
+
+	var functions map[string]FunctionValue
+	if functionCount > 0 {
+		functions = make(map[string]FunctionValue, functionCount)
+	}
```

Comment:
Could early-exit from here
```suggestion
	if functionCount == 0 {
		return nil
	}
	
	functions := make(map[string]FunctionValue, functionCount)
```
```